packages/@lwc/integration-karma/test/template/attribute-class/object-values.spec.js

```diff
@@ -0,0 +1,129 @@
+import { createElement } from 'lwc';
+
+import Dynamic from 'x/dynamic';
+import Reactive from 'x/reactive';
+
+function createDynamicClass(value) {
+    const elm = createElement('x-dynamic', { is: Dynamic });
+    elm.dynamicClass = value;
+    document.body.appendChild(elm);
+
+    return {
+        host: elm,
+        target: elm.shadowRoot.querySelector('div'),
+    };
+}
+
+function testClassNameValue(name, value, expected) {
+    it(name, () => {
+        const { target } = createDynamicClass(value);
+        expect(target.className).toBe(expected);
+    });
+}
+
+function testReactiveClassNameValue(name, setupFn, updateFn, expected) {
+    it(name, () => {
+        const elm = createElement('x-reactive', { is: Reactive });
+        elm.updateDynamicClass(setupFn);
+        document.body.appendChild(elm);
+
+        const target = elm.shadowRoot.querySelector('div');
+
+        elm.updateDynamicClass(updateFn);
+        return Promise.resolve().then(() => {
+            expect(target.className).toBe(expected);
+        });
+    });
+}
+
+// Object class binding is enabled starting from API v61.
+const isObjectBindingEnabled = process.env.API_VERSION >= 61;
+
+describe('type coercion', () => {
+    testClassNameValue('object', {}, isObjectBindingEnabled ? '' : '[object Object]');
+    testClassNameValue('true', true, isObjectBindingEnabled ? '' : 'true');
+    testClassNameValue('false', false, isObjectBindingEnabled ? '' : 'false');
+    testClassNameValue('null', null, isObjectBindingEnabled ? '' : '');
+    testClassNameValue('undefined', undefined, isObjectBindingEnabled ? '' : '');
+    testClassNameValue('number', 1, isObjectBindingEnabled ? '' : '1');
+    testClassNameValue('map', new Map(), isObjectBindingEnabled ? '' : '[object Map]');
+    testClassNameValue('function', function () {}, isObjectBindingEnabled ? '' : 'function () {}');
```

Comment:
Similar to my comment above: I don't see the need to change the current behavior for booleans or numbers.

Granted, some people will probably be tempted to do:

```html
<h1 class={myClass}></h1>
```

... where `myClass` is either `"foo"` or `false`. But in that case, there is little difference between rendering `""` versus `false`, since `false` will probably not match anything. So I would lean towards keeping the existing behavior.

Otherwise, it's just one more thing we have to document in terms of things that may break when you bump your API version.

For objects, map, and functions, I think rendering something like `[object Map]` is obviously bizarre and surprising. For undefined and null, rendering `""` makes sense since they are both nullish.

Reply:
After some thought I take it back. It's fine to render booleans and numbers as `""`.